pkg/resources/model/repository.go

```diff
@@ -63,9 +64,18 @@ func (r LoggingResourceRepository) LoggingResourcesFor(ctx context.Context, logg
 	errs = errors.Append(errs, err)
 
 	watchNamespaces := logging.Spec.WatchNamespaces
+	nsLabelSelector := logging.Spec.WatchNamespaceSelector
 	if len(watchNamespaces) == 0 {
 		var nsList corev1.NamespaceList
-		if err := r.Client.List(ctx, &nsList); err != nil {
+		selector, err := metav1.LabelSelectorAsSelector(nsLabelSelector)
+		if err != nil {
+			errs = errors.Append(errs, errors.WrapIf(err, "error in watchNamespaceSelector"))
+			return
+		}
+		nsListOptions := &client.ListOptions{
+			LabelSelector: selector,
+		}
+		if err := r.Client.List(ctx, &nsList, nsListOptions); err != nil {
```

Comment:
Can we add an if so that we don't prepare a selector if WatchNamespaceSelector is nil?

Reply:
👍 added